pkg/chart/chart.go

```diff
@@ -171,7 +178,31 @@ func NewTesting(config config.Configuration) Testing {
 		directoryLister:  util.DirectoryLister{},
 		chartUtils:       util.ChartUtils{},
 	}
-	return testing
+	if testing.config.Upgrade {
```

Comment:
I don't like this. `NewTesting` is a constructor-like function and, as such, should not contain logic. Neither should it return an error.